optimizely/event_builder.py

```diff
@@ -35,78 +35,95 @@ class BaseEventBuilder(object):
 
   def __init__(self, config):
     self.config = config
-    self.params = {}
 
   @abstractproperty
   class EventParams(object):
     pass
 
-  def _add_project_id(self):
-    """ Add project ID to the event. """
+  def _get_project_id(self):
+    """ Get project ID.
 
-    self.params[self.EventParams.PROJECT_ID] = self.config.get_project_id()
+    Returns:
+      Project ID in the datafile.
+    """
+
+    return self.config.get_project_id()
+
+  def _get_account_id(self):
+    """ Get account ID.
 
-  def _add_account_id(self):
-    """ Add account ID to the event. """
+    Returns:
+      Account ID in the datafile.
+    """
 
-    self.params[self.EventParams.ACCOUNT_ID] = self.config.get_account_id()
+    return self.config.get_account_id()
 
   @abstractmethod
-  def _add_user_id(self, user_id):
-    """ Add user ID to the event.
+  def _get_user(self, user_id):
```

Comment:
This does not make sense as you are passing it `user_id` which is already available

Reply:
Removed this method. And brought logic out into get_common_params method.